x-pack/plugins/entity_manager/server/lib/entities/ingest_pipeline/generate_history_processors.ts

```diff
@@ -140,52 +140,24 @@ export function generateHistoryProcessors(definition: EntityDefinition) {
       },
     },
     {
-      script: {
-        description: 'Generated the entity.id field',
-        source: cleanScript(`
-        // This function will recursively collect all the values of a HashMap of HashMaps
-        Collection collectValues(HashMap subject) {
-          Collection values = new ArrayList();
-          // Iterate through the values
-          for(Object value: subject.values()) {
-            // If the value is a HashMap, recurse
-            if (value instanceof HashMap) {
-              values.addAll(collectValues((HashMap) value));
-            } else {
-              values.add(String.valueOf(value));
-            }
-          }
-          return values;
-        }
-
-        // Create the string builder
-        StringBuilder entityId = new StringBuilder();
+      set: {
+        field: 'entity.id',
+        value: definition.identityFields
+          .map((identityField) => {
+            const field = `entity.identity.${identityField.field}`;
```

Comment:
can we sort the identityFields to make sure it is consistent even if the order of identityFields for whatever reason changes?

Reply:
the pr desc mentions this but looks like it's not implemented yet

Reply:
I think we can expose utility methods that convert the id into an entity, and given that, I would personally prefer sorting the identity fields because we can sort it always when we deserialize, instead of _hoping_ no one changed the order of the identity fields (for whatever reason).

Reply:
In a world where interactions go through an entities client, that should be fine, I'll add the sorting then.

Reply:
I added an issue for this: https://github.com/orgs/elastic/projects/1504/views/1?filterQuery=id&pane=issue&itemId=82919655